tests/acceptance/02_classes/01_basic/common_bundle_select_class_list.cf

```diff
@@ -0,0 +1,35 @@
+# Test that select_class is skipped when the list included is not yet expandable
+body common control {
+      inputs => { "../../default.cf.sub" };
+      bundlesequence  => { default("$(this.promise_filename)") };
+      version => "1.0";
+}
+
+bundle common test 
+{
+  meta:
+      "test_soft_fail" string => "any",
+        meta => { "redmine7482" };
+
+  vars:
+    any::
+      "common_classes" slist => { "common_1", "common_2" };
+
+    common_class_selected::	
+      "class_selected"
+        string => "$(common_classes)",
+        if => "$(common_classes)";
+  
+  classes:
+      # During pre-eval classes are checked first. During that initial pass the
+      # common_classes list is undefined, and so the promise should be skipped.
```

Comment:
This is not quite true unfortunately. It won't be evaluated during first pre-evaluation round but there are multiple and should be evaluated in subsequent.